docs/resources/task.md

```diff
@@ -0,0 +1,221 @@
+# Task Resource
+
+Task, consisting of one or more cloud machines that run a given `script` upon
+completion or `timeout`, sharing a common network and a storage `directory`.
+
+## Example Usage
+
+```hcl
+resource "iterative_task" "task" {
+  name  = "example"
+  cloud = "aws"
+
+  environment = {GREETING = "Hello, world!"}
+  directory   = "${path.root}/results"
+
+  script = <<-END
+    #!/bin/bash
+    echo "$GREETING" | tee $(uuidgen)
+  END
+}
+```
+
+## Argument Reference
+
+### Required
+
+- `name` - (Required) Task name.
+- `cloud` - (Required) Cloud provider to run the task on; valid values are `aws`, `gcp`, `az` and `k8s`.
+- `script` - (Required) Script to run; must begin with a valid [shebang](<https://en.wikipedia.org/wiki/Shebang_(Unix)>).
+
+### Optional
+
+- `region` - (Optional) Cloud region/zone to run the task on.
+- `machine` - (Optional) See [Machine Types](https://registry.terraform.io/providers/iterative/iterative/latest/docs/resources/task#machine-types) below.
+- `disk_size` - (Optional) Size of the ephemeral machine storage.
+- `spot` - (Optional) Spot instance price. `-1`: disabled, `0`: automatic price, any other positive number: fixed price.
+- `image` - (Optional) Machine image to run the task with.
+- `parallelism` - (Optional) Number of machines to be launched in parallel.
+- `directory` - (Optional) Local directory to synchronize.
+- `environment` - (Optional) Environment variables to run the task with.
```

Comment:
```suggestion
- `environment` - (Optional) Environment variables to run the task with; empty values will cause the provider to retrieve them from the local environment.
```

Reply:
For example, `PATH` below will be loaded from the local environment.

```hcl
environment = {
  PATH = ""
  CUSTOM = "VALUE"
}
```

Reply:
@casperdcl, can you please bless this with your descriptive clarity?

Reply:
aha! https://github.com/iterative/terraform-provider-iterative/pull/237#discussion_r747892116 :)

Reply:
vis. https://github.com/iterative/terraform-provider-iterative/pull/237#discussion_r747897237